R/install-git.r

```diff
@@ -69,3 +76,44 @@ remote_metadata.git_remote <- function(x, bundle = NULL, source = NULL) {
     RemoteSha = sha
   )
 }
+
+remote_sha.git_remote <- function(remote, ...) {
+  res <- system(
```

Comment:
https://github.com/ropensci/git2r/pull/172 adds `ls_remote()` to git2r, which exposes this functionality to us.

This PR also uses `git archive` to download the file for plain git remotes.  I imagine git2r doesn't have this either.  I think it is lower priority though as plain git is probably used less frequently than github or bitbucket and people will probably already have a git client installed in that case (or just always install the package as is the current behavior).